### PR TITLE
reference: add some explanation about _tidb_rowid

### PR DESCRIPTION
**Pull request by Hangzhi**
### What is changed, added or deleted? (Required)

This is a translation from docs-cn. A sql example is added.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/2569
- Other reference link(s):

### CODE REVIEW

**Comment by TomShawn** [comment]
@Hangzhi Thanks for your contribution!

File: reference/mysql-compatibility.md [hunk 1]
```diff
@@ -68,6 +68,29 @@ The operations are executed as follows:
 
 Also, starting from TiDB 2.1.18 and 3.0.4, TiDB supports using the system variable `tidb_allow_remove_auto_inc` to control whether the `AUTO_INCREMENT` property of a column is allowed to be removed by executing  `ALTER TABLE MODIFY` or `ALTER TABLE CHANGE` statements. It is not allowed by default.
 
+> **Note:**
+>
+> If the primary key is not specified, `_tibd_rowid` will be used to identify the row in TiDB. Allocation of this variable and auto-increment column will the same allocator. If the auto-increment column is specified as the primary key, then the column will be used to identify rows. So, there will be the following situation.
```
Comment (yikeke) [inline reference/mysql-compatibility.md:73]:
```suggestion
> If the primary key is not specified, TiDB uses the `_tibd_rowid` column to identify rows. The values of the `_tibd_rowid` column and the auto-increment column (if there is) are assigned by the same allocator. If the auto-increment column is specified as the primary key, then TiDB uses this column to identify rows. Therefore, there might be the following situations.
```

Comment (AilinKid) [reply reference/mysql-compatibility.md:73]: > 

LGTM

Comment (yikeke) [reply reference/mysql-compatibility.md]: My suggestion is small so I'll apply it now to allow this PR to be merged. Thanks for your contribution! @Hangzhi

**Pull request merged**